### PR TITLE
Show tree with phantom

### PR DESCRIPTION
**Pull request by SeveNChaK**

### CODE REVIEW

File: SublimeQuip.py [hunk 1]
```diff
@@ -2,43 +2,100 @@
 import sublime_plugin
 from .src.providers import quip_provider
 from .CurrentManager import CurrentManager
+from .TabsManager import TabsManager
+from .TabsManager import FILE_TREE_TAB_ID
+
+COMMAND_OPEN_DOCUMENT = "open_recent_document"
```
Comment (iOctav) [inline SublimeQuip.py:14]: Let's move all constants to separate file

File: SublimeQuip.py [hunk 2]
```diff
@@ -2,43 +2,100 @@
...
+			if tabs_manager.contains_tab(thread_id):
+				view = tabs_manager.get_tab(thread_id)
+				self.window.focus_view(view)
+			else:
+				view = self.window.new_file()
+				tabs_manager.add_tab(thread_id, view)
+			view.run_command(COMMAND_GET_SELECTED_DOCUMENT, args)
+		else:
+			view = self.window.new_file()
+			view.run_command(COMMAND_GET_RANDOM_DOCUMENT)
```
Comment (iOctav) [inline SublimeQuip.py:44]: Let's use recent keyword instead of random

File: SublimeQuip.py [hunk 3]
```diff
@@ -2,43 +2,100 @@
...
 
 class ShowFileTreeCommand(sublime_plugin.WindowCommand):
 	def run(self):
-		view = self.window.new_file()
-		view.run_command("printquipfiletree")
+		if not tabs_manager.contains_tab(FILE_TREE_TAB_ID):
```
Comment (iOctav) [inline SublimeQuip.py:49]: Is it really necessary pass constant from TabsManager class into TabsManager method?

File: SublimeQuip.py [hunk 4]
```diff
@@ -2,43 +2,100 @@
...
+			view = self.window.new_file()
+			tabs_manager.add_tab(FILE_TREE_TAB_ID, view)
+		else:
+			view = tabs_manager.get_tab(FILE_TREE_TAB_ID)
+			self.window.focus_view(view)
+		view.run_command(COMMAND_PRINT_QUIP_FILE_TREE)
 
-class InsertrandomdocumenthtmlCommand(sublime_plugin.TextCommand):
+class InsertRandomDocumentHtmlCommand(sublime_plugin.TextCommand):
```
Comment (iOctav) [inline SublimeQuip.py:58]: The same, let's replace random to recent

File: SublimeQuip.py [hunk 5]
```diff
@@ -2,43 +2,100 @@
...
 		id = thread_ids.pop()
 		current.add(view=self.view, thread=id)
 		self.view.insert(edit, 0, quipprovider.get_document_content(id))
 
-class Printquipfiletree(sublime_plugin.TextCommand):
-	def __print_tree(self, tree_node, prefix):
+class InsertSelectedDocumentCommand(sublime_plugin.TextCommand):
+	def run(self, edit, **args):
+		doc_id = args[KEY_THREAD_ID]
+		assert doc_id is not None
```
Comment (iOctav) [inline SublimeQuip.py:70]: Is it using for debugging?

File: SublimeQuip.py [hunk 6]
```diff
@@ -2,43 +2,100 @@
...
+		quipprovider = quip_provider.QuipProvider()
+		current.add(view=self.view, thread=doc_id)
+		self.view.replace(edit, self.view.visible_region(), quipprovider.get_document_content(doc_id))
```
Comment (iOctav) [inline SublimeQuip.py:72]: Does visible_region return full text of file or only part which is seen on my screen?

File: TabsManager.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+from sublime import *
+from sublime_plugin import *
+
+FILE_TREE_TAB_ID = "file_tree_tab_id"
+
+class TabsManager:
+
+    def __init__(self):
+        self.visible_tabs = dict()
```
Comment (iOctav) [inline TabsManager.py:9]: opened_tabs is more self-documenting name

**Review by iOctav — APPROVED** [review]

File: SublimeQuip.py (hunk 4 above)
Comment (SeveNChaK) [reply SublimeQuip.py:58]: Это ты же сам написал random)

File: SublimeQuip.py (hunk 6 above)
Comment (SeveNChaK) [reply SublimeQuip.py:72]: Интересный вопрос, я сильно не вдавался, проверю

File: SublimeQuip.py (hunk 4 above)
Comment (iOctav) [reply SublimeQuip.py:58]: Yeap, it was example with awesome name

File: SublimeQuip.py [hunk 8]
```diff
@@ -1,11 +1,16 @@
 import sublime
 import sublime_plugin
 import json
+import os
 from .src.providers import quip_provider
 from .CurrentManager import CurrentManager
 from .TabsManager import TabsManager
 from .TabsManager import FILE_TREE_TAB_ID
 
+CACHE_DIRECTORY = sublime.cache_path() + '/QUIP'
```
Comment (iOctav) [inline SublimeQuip.py]: '/SublimeQuip'

Comment (iOctav) [reply SublimeQuip.py]: double quotes

**Review by iOctav — CHANGES_REQUESTED** [review]

**Comment by iOctav** [comment]
If you open document twice after closing, content is doubled.

**Comment by mandarin1029** [comment]
https://github.com/SPBSTU-OrangeTeam/SublimeQuip/pull/5#issuecomment-739566086
> If you open document twice after closing, content is doubled.

Fixed 7db8920

File: SublimeQuip.py (hunk 8 above)
Comment (mandarin1029) [reply SublimeQuip.py]: @iOctav 
458750a

File: SublimeQuip.py [hunk 9]
```diff
@@ -7,7 +7,7 @@
 from .TabsManager import TabsManager
 from .TabsManager import FILE_TREE_TAB_ID
 
-CACHE_DIRECTORY = sublime.cache_path() + '/QUIP'
+CACHE_DIRECTORY = sublime.cache_path() + "/SublimeQuip"
```
Comment (iOctav) [inline SublimeQuip.py]: revert please, we will rename plugin as Quip

**Review by iOctav — APPROVED** [review]

File: SublimeQuip.py (hunk 9 above)
Comment (mandarin1029) [reply SublimeQuip.py]: 5e5ff0b

**Pull request merged**